src/components/ui/toolbox.ts

```diff
@@ -337,34 +387,42 @@ export default class Toolbox extends Module<ToolboxNodes> {
    * @param {string} toolName - Tool name
    */
   private insertNewBlock(toolName: string): void {
-    const { BlockManager, Caret } = this.Editor;
-    const { currentBlock } = BlockManager;
+    const currentBlockIndex = this.api.blocks.getCurrentBlockIndex();
+    const currentBlock = this.api.blocks.getBlockByIndex(currentBlockIndex);
 
-    const newBlock = BlockManager.insert({
-      tool: toolName,
-      replace: currentBlock.isEmpty,
-    });
+    if (!currentBlock) {
+      return;
+    }
+
+    /**
+     * On mobile version, we see the Plus Button even near non-empty blocks,
+     * so if current block is not empty, add the new block below the current
+     */
+    const index = currentBlock.isEmpty ? currentBlockIndex : currentBlockIndex + 1;
+
+    const newBlock = this.api.blocks.insert(
+      toolName,
+      undefined,
+      undefined,
+      index,
+      undefined,
+      currentBlock.isEmpty
```

Comment:
We should go with object here

Reply:
I've decided not to change the interface of the existed API method. Should I create overload for that?